packages/runtime-html/src/bindable.ts

```diff
@@ -317,3 +321,59 @@ function apiTypeCheck() {
     .add('prop').primary();
 }
 /* eslint-enable @typescript-eslint/no-unused-vars,spaced-comment */
+
+/** @internal */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+declare namespace Reflect {
+  function getMetadata(metadataKey: any, target: any, propertyKey?: string | symbol): any;
+}
+
+export function coercer(target: Constructable<unknown>, property: string, _descriptor: PropertyDescriptor): void {
+  Coercer.define(target, property);
+}
+
+const Coercer = {
+  key: getAnnotationKeyFor('coercer'),
+  define(target: Constructable<unknown>, property: string) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+    defineMetadata(Coercer.key, ((target as any)[property] as InterceptorFunc).bind(target), target);
+  },
+  for(target: Constructable<unknown>) {
+    return getOwnMetadata(Coercer.key, target) as InterceptorFunc;
+  }
+};
+
+function getInterceptor(prop: string, target: Constructable<unknown>, def: PartialBindableDefinition = {}) {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const type: PropertyType | null = def.type ?? Reflect.getMetadata('design:type', target, prop) ?? null;
+  if (type == null) { return noop; }
+  let coercer: InterceptorFunc;
+  switch (type) {
+    case Number:
+    case Boolean:
+    case String:
+    case BigInt:
+      coercer = type as InterceptorFunc;
+      break;
+    default: {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+      const $coercer: InterceptorFunc = (type as any).coercer as InterceptorFunc;
+      coercer = typeof $coercer === 'function'
+        ? $coercer.bind(type)
+        : (Coercer.for(type as Constructable) ?? noop);
+      break;
+    }
+  }
+  return coercer === noop
+    ? coercer
+    : createCoercer(coercer, def.nullable);
+}
+
+function createCoercer<TInput, TOutput>(coercer: InterceptorFunc<TInput, TOutput>, nullable: boolean | undefined): InterceptorFunc<TInput, TOutput> {
+  return function (value: TInput, coercionConfiguration: ICoercionConfiguration | null): TOutput {
+    if (!(coercionConfiguration?.enableCoercion ?? false)) return value as unknown as TOutput;
```

Comment:
```suggestion
    if (!coercionConfiguration?.enableCoercion) return value as unknown as TOutput;
```
I'm not sure which is better between ensuring boolean or just use the language feature, micro bench but if we look at code size, probably `!` is sufficient